Library/ViewModels/RewardCellViewModel.swift

```diff
@@ -57,21 +57,31 @@ RewardCellViewModelOutputs {
     }
     let projectAndReward = Signal.zip(project, reward)
 
-    self.conversionLabelHidden = project.map {
-      !needsConversion(projectCountry: $0.country, userCountry: AppEnvironment.current.config?.countryCode)
+    self.conversionLabelHidden = project.map { p in
+      p.stats.currentCurrency
+        .map { !needsConversion(projectCountry: p.country, currentCurrency: $0) }
+        ?? !needsConversion(projectCountry: p.country,
+                            userCountry: AppEnvironment.current.config?.countryCode)
     }
 
     self.conversionLabelText = projectAndRewardOrBacking
       .filter { p, _ in
-        needsConversion(projectCountry: p.country, userCountry: AppEnvironment.current.config?.countryCode)
+        p.stats.currentCurrency
+          .map { needsConversion(projectCountry: p.country, currentCurrency: $0) }
+          ?? needsConversion(projectCountry: p.country,
+                             userCountry: AppEnvironment.current.config?.countryCode)
       }
       .map { project, rewardOrBacking in
+        let (country, rate) = zip(
+          project.stats.currentCurrency.flatMap(Project.Country.init(currencyCode:)),
```

Comment:
Compiler clever enough here for `.flatMap(.init(currencyCode:))`?

Reply:
I don't think so. Since function isn't its own type and this is point-free.